packages/slate-editor/src/extensions/coverage/lib/getCoverageImageUrl.ts

```diff
@@ -0,0 +1,22 @@
+import type { CoverageEntry } from "@prezly/sdk";
+import type { UploadedImage } from "@prezly/uploadcare";
+import { UploadcareImage } from "@prezly/uploadcare";
+
+const IMAGE_WIDTH = 580 * 2;
+
+export function getCoverageImageUrl(coverage: CoverageEntry): string | null {
+    if (coverage.attachment_oembed && coverage.attachment_oembed.thumbnail_url) {
+        return coverage.attachment_oembed.thumbnail_url;
+    }
+
+    if (coverage.attachment) {
+        try {
+            const image = UploadcareImage.createFromPrezlyStoragePayload(coverage.attachment as UploadedImage);
```

Comment:
I've noticed the `attachment` is actually of type `UploadedFile` so the previous condition where we checked if it's an `UploadcareImage` failed, because the object doesn't have `effects` and other image properties, so thumbnails were not working at all.

I guess we still need that check because `attachment` can be a non-image file in which case it's not possible to display however I wonder why we're not using the `UploadedImage` structure for image files.

Is this a bug in the API? 🤔

Reply:
But it's `attachment`, not cover image. It can be a PDF file, MP3 or a video file.

Reply:
Yeah, I've realised that, but the problem is that we were trying to render the `attachment` as a thumbnail if it's an image, but even if you upload an image itself, the API will return the payload without the `effects` and other image-specific properties so the condition will never be fulfilled.

Reply:
> but even if you upload an image itself, the API will return the payload without the effects and other image-specific properties so the condition will never be fulfilled.

Ah, that... In that case yeah, we can consider this as a bug in the API.

Reply:
Checked with @pestaa and turns out frontend was sending the `UploadcareFile` payload instead of `UploadcareImage` payload if attachment was an image 🥲 

So any past uploaded attachments will not have the image-specific properties until https://github.com/prezly/prezly/pull/15053 is deployed.

Either way, I think this should still work as the `UploadcareImage` component should throw an error if it's not possible to construct the object from the given payload.

I'll test this to make sure.